internal/authentication/azure/auth.go

```diff
@@ -103,7 +136,9 @@ func (s EnvironmentSettings) GetAzureEnvironment() (*azure.Environment, error) {
 // GetTokenCredential returns an azcore.TokenCredential retrieved from, in order:
 // 1. Client credentials
 // 2. Client certificate
-// 3. MSI
+// 3. MSI with 1 second timeout
+// 4. CLI credential with 5 second timeout
```

Comment:
What if we moved CLI credentials before MSI? THis way you wouldn't need to try MSI twice with different timeouts. I think it's acceptable to have CLI have priority over MSI.

Reply:
It would not be ideal to delay startup in production because you're waiting for an azure CLI timeout, does these fail quickly or do they do HTTP requests that eventually timeout when an MSI endpoint isnt available?

Reply:
I'm actually not even sure why the CLI's credential check needs a context. All methods in the Azure SDKs that use a context are making network calls, but the CLI credentials are stored on a local file (unless maybe the method is making a call to exchange a refresh token?)

Another option could be to trigger both these checks in parallel (each on its own goroutine) and then the first one that returns (successfully) wins. Kind of like Promise.race in JavaScript!

Reply:
I just followed what the DefaultAzureCredential essentially does. All this code is pretty much copied from the `azidentity` package.

Reply:
@onionhammer @ItalyPaleAle 

The `DefaultAzureCredential` tries Managed Identity with a time out of 1 (one) second before trying the CLI Credential. This will usually be no problem. However, in the case this does time out I added an indefinite attempt to connect after the Azure CLI Credential. I think this is a good compromise that also matches what the DefaultAzureCredential does - but with the additional Managed Identity Retry without the 1 second timeout.

Reply:
Oh, if this is what the Azure team is doing too, then I have no more concerns.